relay-pattern/src/typed.rs

```diff
@@ -171,4 +303,78 @@ mod tests {
         let pattern: TypedPattern<CaseInsensitive> = TypedPattern::new("*[rt]x").unwrap();
         assert_eq!(serde_json::to_string(&pattern).unwrap(), r#""*[rt]x""#);
     }
+
+    #[test]
+    fn test_patterns_default() {
+        let patterns: TypedPatterns = TypedPatterns::builder()
+            .add("*[rt]x".to_owned())
+            .unwrap()
+            .add("foobar".to_owned())
+            .unwrap()
+            .take();
+        assert!(patterns.is_match("f/o_rx"));
+        assert!(patterns.is_match("foobar"));
+        assert!(!patterns.is_match("Foobar"));
+    }
+
+    #[test]
+    fn test_patterns_case_insensitive() {
+        let patterns: TypedPatterns<CaseInsensitive> = TypedPatterns::builder()
+            .add("*[rt]x".to_owned())
+            .unwrap()
+            .add("foobar".to_owned())
+            .unwrap()
+            .take();
+        assert!(patterns.is_match("f/o_rx"));
+        assert!(patterns.is_match("f/o_Rx"));
+        assert!(patterns.is_match("foobar"));
+        assert!(patterns.is_match("Foobar"));
+    }
+
+    #[test]
+    #[cfg(feature = "serde")]
+    fn test_patterns_deserialize() {
+        let pattern: TypedPatterns<CaseInsensitive> =
+            serde_json::from_str(r#"["*[rt]x","foobar"]"#).unwrap();
+        assert!(pattern.is_match("foobar_rx"));
+        assert!(pattern.is_match("FOOBAR"));
+    }
+
+    #[test]
+    #[cfg(feature = "serde")]
+    fn test_patterns_deserialize_err() {
+        let r: TypedPatterns<CaseInsensitive> =
+            serde_json::from_str(r#"["[invalid","foobar"]"#).unwrap();
```

Comment:
FWIW failing deserialization would also be acceptable behavior here.

Reply:
Yes, tried to describe this in the PR description. I chose to ignore it for now because it mimics `glob3` and makes the transition much easier.